### ISSUE TITLE
Unable to Perform CCLA using EasyCLA API when a parent project CLA Groups exists

### ISSUE THREAD

**Issue by jarias-lfx** [communitybridge/easycla#3773]
## Summary

When trying to perform the Corporate CLA signature using EasyCLA API (using the GitHub flow) for a subproject CLA Group, if the project had a parent project CLA Group, fails with a 400 error.

## Background

Recent EasyCLA feature where a child project CLA Group can be created when there is an existing parent project CLA Group. When using EasyCLA API (GitHub flow). After selecting the organization and choose yes for the CLA Manager option. The system responds with a 400 error
```
{"Code":"400","Message":"EasyCLA - 400 Bad Request - Creating cla manager failed for Foundation SFID: a092M00001JwBlEQAV  - error: project already signed","x-request-id":"97417280-a950-11ed-ac9c-19f931db2711"}
```
The error is referencing the parent project CLA Group.

## Expected behavior

When choosing yes for the CLA Manager option, the system should assign the cla-manager-designee role and proceed with the API flow: https://docs.linuxfoundation.org/lfx/easycla/v2-current/contributors/corporate-contributor#if-company-has-not-signed

## Screenshots

![image](https://user-images.githubusercontent.com/93140557/218120984-d365628a-ff93-4f6b-8a96-ada63e6272da.png)

## Environment

Please complete the following information:

* Environment:
  - [*] ALL
  - [ ] DEV
  - [ ] STAGING
  - [ ] PROD 
* Browser:
  - [*] Chrome/Brave
  - [ ] Firefox
  - [ ] Opera
  - [ ] Vivaldi
  - [ ] LibreWolf
  - [ ] SRware Iron
  - [ ] Dissenter
  - [ ] Slimjet
  - [ ] Midori
  - [ ] Edge
  - [ ] Lynx
* Version: v1.0.XX

## Additional Context

Tested the flow using different GitHub and LFID accounts. One attached to The Linux Foundation organization, the other attached to another organization. This to confirm the issue is not isolated to The Linux Foundation organization. On both scenarios the flow fails with the error reported.

Additionally, this was tested by different users with the same result.

## Steps to Reproduce

1- Access https://contributor.easycla.lfx.linuxfoundation.org/#/cla/project/a49a37c7-1506-4cc8-9080-5bc34d675f85/user/04581041-79ab-401b-96be-6b44c85c3a46
2- Select Proceed as Corporate contributor
3- Search for the organization name you are contributing with
4- Select proceed
5- Select yes on the CLA Manager confirmation option

**Comment by dealako** [comment]
I investigated this a bit. Here is what I found:

This occurs when the CLA Group is created when a parent (umbrella) CLA group exists. When the create CLA group API call is made, the `input.FoundationSfid` value is the parent (FINOS parent in this case), and the `input.ProjectSfidList[0]` is the current FINOS child project (CDM or FDC3 in this case).  This set of parameters ultimately leads to a table entry in the `cla-prod-projects-cla-groups` table which has the `input.FoundationSfid` provided value (FINOS parent) in the `foundation_sfid` column.

For this setup/configuration, since it is a child CLA group under the parent that already has a CLA Group, we **do not** want to use the parent (FINOS) SFID as the value in the `cla-prod-projects-cla-groups` table. Instead, this table column should have the current child project SFID value (CDM or FDC3 in this case).

**Comment by dealako** [comment]
@thakurveerendras 

Please assist by testing this scenario. Work with @jarias-lfx to confirm the use cases.

**Comment by thakurveerendras** [comment]
Issue is still reproducible as reported in main comments on dev site 

Setup : 

- Parent project :  CCLAs singed : https://lfx.dev.platform.linuxfoundation.org/foundation/a092h000004x5tVAAQ/cla
- Child project : CCLAs not singed : https://lfx.dev.platform.linuxfoundation.org/foundation/a092h000004x5tVAAQ/project/a092h000004x5taAAA/cla

Steps: 

1. Go to gated Easycla PR : https://github.com/earthChildOrg/Repo04/pull/1
2. Click on Not covered
3. Select Employee Acknowledgement
4. Search & selected the company
5. Click on Procced button

Actual results :Below error appears
 "EasyCLA - 400 Bad Request - Creating cla manager failed for Foundation SFID: a092h000004x5tVAAQ - error: project already signed"

Refer video clip : 
[3773.webm](https://user-images.githubusercontent.com/75417507/218748542-3cffca21-596d-4d4e-afeb-ce7103352c58.webm)

**Comment by nickmango** [comment]
@dealako in addition to the identified issue youd resolved theres a process that involves creating cla_manager_designee role for the project. In this case foundationSFID is used for the operation which should not be the case. Probably logic checked if project is signed at foundation level and used the foundationSFID for the role assignment.

**Comment by jarias-lfx** [comment]
Adding to this issue: 
Discovered that changing the foundation_sfid value from the parent project to the child project (which fixes the corporate console signature process issue), In the PCC, the CLA Groups are not longer being displayed, it seems the PCC uses this value to include the group list.

Screenshot after replacing the foundation_sfid from the parent project to the child project:
![image](https://user-images.githubusercontent.com/93140557/218826294-6e499076-9358-452d-83e0-c610b4324fe6.png)

Screenshot when the three CLA Groups share the same foundation_sfid (which causes this issue):
![image](https://user-images.githubusercontent.com/93140557/218826563-5865cece-6cea-4f99-b628-88808c373daf.png)

**Comment by thakurveerendras** [comment]
@nickmango , Issue is still reproducible as reported on dev site 

- Refer PR : https://github.com/earthChildOrg/Repo04/pull/2
- Organization : Infosys limited 

![image](https://user-images.githubusercontent.com/75417507/220014224-e43a2535-d9a2-4a59-8bf2-014875f1e894.png)

**Comment by thakurveerendras** [comment]
Now "Request Failed" prompt appears when apply above steps 

![image](https://user-images.githubusercontent.com/75417507/220157584-6e57f9f8-c394-47f4-8a91-3a9e22cb5f86.png)

**Comment by nickmango** [comment]
@thakurveerendras kindly reverify this

**Comment by thakurveerendras** [comment]
Hello @nickmango 
Tested user on same project with two different user 

1. User 1 : Organization : Infosys Limited 

- Able to singed cclas properly on child project via contributor console when ccla exist on parent project

3. User 2 : Organization : Indigo

- Nothing happen after click on procced button at "Configuring CLA Manager Settings"
Refer video clip : 

https://user-images.githubusercontent.com/75417507/220611675-52d80c4e-d9bd-47b9-b2a7-9d44a78e5c2d.mp4


https://user-images.githubusercontent.com/75417507/220611267-064f67d8-d14d-452f-9b7c-05dbc272b8fc.mp4

**Comment by thakurveerendras** [comment]
@nickmango , Kindly refer steps for above failed case

1. Login PCC with vthakur.lflegal user
2. Search & open SUN project 
3. Go to Parent project PR & singed cclas via contributor console
4. Go to child project PR & try to complete cclas via contributor console 

Actual results : Nothing happen after click on procced button at "Configuring CLA Manager Settings"
Refer video clip added on above comments

**Comment by jarias-lfx** [comment]
After https://github.com/communitybridge/easycla/tree/v2.0.125 release in PROD. I was able to perform a CCLA signature process from start to finish in FDC3 CSL using a test organization. The test organization has a previous signature with FINOS Apache CCLA. No issues seen during the process. After the signature I was able to add an approval criteria reported on https://github.com/communitybridge/easycla/issues/3779
